pontoon/base/utils.py

```diff
@@ -541,3 +541,27 @@ def readonly_exists(projects, locale):
     return ProjectLocale.objects.filter(
         project__in=projects, locale=locale, readonly=True,
     ).exists()
+
+
+def filter_users_by_project_visibility(project, locale, users):
+    """
+    Filter users that can access the project and return their ids.
+    """
+    from .models import User, ProjectLocale
+
+    if project.visibility == "public":
+        return users
+
+    return (
+        User.objects.filter(pk__in=users)
+        .filter(
+            groups__in=[
+                ProjectLocale.objects.get(
+                    project=project, locale=locale
+                ).translators_group,
+                locale.translators_group,
+                locale.managers_group,
+            ]
```

Comment:
I don't understand this filter in the context of bug 1627877.

In fact, I think we should remove this entire function. If the Admin submits a comment in a Private project, that probably means they forgot to make the project accessible to the recipient. Without the check recipient will at least be able to complain about not being able to see the comment/project.

If anything, we should prevent submission of comments on Private projects, but I think that's slightly overengineered at this stage, since only Admins would be able to make such mistake.